### PR TITLE
Allow additional refstate transitions for ball placement

### PR DESCRIPTION
**Pull request by AlexanderDanzer**
For the autoref implementation I would like to be able to perform
some additional refstate switches.
These concern the ball placement state and eliminate the
need of a detour via the stop state.

### CODE REVIEW

File: gamecontroller.cc [hunk 1]
```diff
@@ -247,10 +248,11 @@ bool GameController::can_set_command(SSL_Referee::Command command) const {
 			// You can award goals whenever you are stopped.
 			return is_stopped;
 
+		// You can ask for ball placement whenever you are stopped or the other team failed to place the ball
 		case SSL_Referee::BALL_PLACEMENT_YELLOW:
+			return is_stopped || ref.command() == SSL_Referee::BALL_PLACEMENT_BLUE;
 		case SSL_Referee::BALL_PLACEMENT_BLUE:
-			// You can ask for ball placement whenever you are stopped.
-			return is_stopped;
+			return is_stopped || ref.command() == SSL_Referee::BALL_PLACEMENT_BLUE;
```
Comment (Hawk777) [inline gamecontroller.cc]: This line should compare to BALL_PLACEMENT_YELLOW, not BALL_PLACEMENT_BLUE.

Comment (AlexanderDanzer) [inline gamecontroller.cc]: I corrected it and did a forced push. Hope that's ok for this small change.

**Comment by Hawk777** [comment]
If we allow a transition directly from placement to direct or indirect free kick, then I think we should also allow a transition directly from placement to prepare kickoff or prepare penalty. Do you agree?

**Comment by AlexanderDanzer** [comment]
I added transitions from ball placement to penalty and kickoff prepare in a second commit.

**Pull request merged**